odl/contrib/README.md

```diff
@@ -0,0 +1,85 @@
+# The `odl.contrib` sub-package
+
+This is a special place for user-contributed modules. Its intention is to provide a "fast lane" for contributors to include code that is
+
+- outside the main ODL library,
+- experimental and untested,
+- unfinished or still subject to major breaking changes or
+- "not quite there yet" in terms of documentation, testing or completeness.
+
+The general idea is that users get their code quickly merged without having to go through lengthy code reviews.
+
+**Note:** If you consider your code to be useful for the main library, please consider filing a pull request anyway. The maintainers try to be as helpful as possible in their reviews and make suggestions, so that getting the code ready for merging doesn't take too long.
+
+Here are some guidelines for contributors regarding the scope and the rules of this platform.
+
+
+## Scope
+
+**Yes:**
+
+- Generic code that would fit into main ODL but is still experimental
+- Special-purpose code that is not generic enough to be a fit for main ODL (e.g. [figures of merit](https://github.com/odlgroup/odl/pull/1018) or application-specific code)
+- Extended, more complex examples than [the ones in main ODL](https://github.com/odlgroup/odl/tree/master/examples)
+
+**No:**
+
+- Code for specialized algorithms written specifically for scientific papers
+- Large files like images, datasets etc.
+- Code under licenses other than MPL-2.0
+
+
+## Procedure
+
+Since you do not have write access to the target location in the ODL repository, do the initial work on the code on a personal fork. Within this fork, add your contribution to the `odl/contrib` directory. When the code is ready, the steps are as follows:
+
+- Submit a pull request. If you would like an in-depth review by a maintainer, please indicate that in writing or by choosing a reviewer on the Github pull request page.
+- A reviewer will check the pull request as a whole and the commit history (and optionally the code).
+- If no in-depth review was asked for, the reviewer will simply merge the PR if everything is okay. Otherwise, merging is delayed until you indicate that you have modified the code according to review.
+
+That's all.
+
+
+## Commit history
+
+Please keep the commit history as short as possible. Consider squashing commits of addition/removal cycles or work-in-progress incremental commits.
+
+
+## Package layout
+
+All code goes into the `odl/contrib` subdirectory. If a contribution consists in a single Python module `mymodule.py`, it can be added directly to that directory. Otherwise, a subdirectory `mylib` should be created that contains all code, i.e., *modules*, *examples* and *tests*.
```

Comment:
Disagree. I'd say they all should go into a folder, mostly for future extensibility and improved compartamentalization.